### DISCUSSION TITLE
Flag/Check: Access Key

### DISCUSSION THREAD

**Discussion by emmapeel2** [WeblateOrg/weblate#10319] [Ideas]
In our project we have several apps that use Access Keys. This keys help on the browsing and accessibility of the applications, and as opposed to shortcuts, can be localized.

But they are very confusing for translators!

For example on this access key: https://hosted.weblate.org/translate/tor/tor-browser/tb-cryptosafetypromptproperties/en/?checksum=2b68b0583089f30c I would like to add the flag: `access_key_for:cryptoSafetyPrompt.secondaryAction` and have weblate fail if the character I am adding is not present at the translation for the Key cryptoSafetyPrompt.secondaryAction (which in this case is the previous string: https://hosted.weblate.org/translate/tor/tor-browser/tb-cryptosafetypromptproperties/en/?checksum=5e55980a7d1f113a)

This way translators will realize what Access Keys are meant for, the difference between AccessKeys and Shortcuts, what do they trigger in the interface, etc.

**Comment by nijel** [comment]
First, I've configured variants for this component so that `cryptoSafetyPrompt.secondaryAction` and `cryptoSafetyPrompt.secondaryActionAccessKey` are grouped together. For example: https://hosted.weblate.org/translate/tor/tor-browser/tb-cryptosafetypromptproperties/en/?checksum=5e55980a7d1f113a#variants

See https://docs.weblate.org/en/latest/devel/translations.html#variants for more info on that.

As for the check, maybe I'm missing something, but the linked string is “Dismiss Warning”, while the access key is “B”, so it is wrong here in the source language?

**Reply by emmapeel2** [reply]
Yes, not sure how it works for this string in particular, but the button says “Dismiss Warning (B)” (See the screenshot). Maybe this example was not the best.

I didn't knew the variants were also about keys, I thought they were a glossary thing, thanks for explaining that.

**Reply by nijel** [reply]
Variants are a generic way to group strings together.
